### PR TITLE
More user-friendly RSS URL generation with extra parameters

### PR DESCRIPTION
**Pull request by TomAbel**

### CODE REVIEW

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 1]
```diff
@@ -327,6 +325,85 @@
                 </div>
               </div>
             </div>
+            <!-- RSS -->
+            <div id="section4" class="tab-pane fade in">
+              <div class="form-group">
+                <label class="col-sm-2 control-label" for="rss-category"><?php echo $rss_category_title ?></label>
+                <div class="col-sm-10">
+                  <select name="smaily_for_opencart_rss_category" id="rss-category" class="form-control smaily-rss-options">
+                    <option value=''>All Products</option>
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:334]: Formatting: mixing single and double quotes in HTML.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 2]
```diff
@@ -327,6 +325,85 @@
...
+                    <?php
+                    foreach ($rss_categories as $category) {
+                      $selected = is_array($rss_category) && in_array($category['category_id'], $rss_category) ? 'selected' : '';
+                      echo("<option value=$category[category_id] $selected>$category[name]</option>");
+                    }
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:339]: Use pattern described here https://github.com/sendsmaily/smaily-opencart-module/pull/94#discussion_r482091656 to rework this code block.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 3]
```diff
@@ -327,6 +325,85 @@
...
+                  <select name="smaily_for_opencart_rss_sort_by" id="rss-sort-by" class="form-control smaily-rss-options">
+                   <?php
+                    // All available options
+                    $sort_options = [
+                      'pd.name' => $sort_name,
+                      'p.model' => $sort_model,
+                      'p.price' => $sort_price,
+                      'p.status' => $sort_status,
+                      'p.sort_order' => $sort_order
+                    ];
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:357]: Provide this mapping from the controller.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 4]
```diff
@@ -327,6 +325,85 @@
...
+                    // Add options for select.
+                    foreach ($sort_options as $sort_code => $sort_name) {
+                      $selected = $rss_sort_by === $sort_code ? 'selected' : '';
+                      echo("<option value='$sort_code' $selected>$sort_name</option>");
+                    }
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:362]: Use the same pattern here, as with category field.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 5]
```diff
@@ -505,6 +582,35 @@
...
+        parameters.limit = rss_limit;
+      }
+      $('#smaily-rss-feed-url').html(rss_url_base + $.param(parameters));
+    });
   });
 })(jQuery);
 </script>
+<script type="text/javascript">
+    var smaily_rss_url_base = '<?php echo $smaily_rss_url_base; ?>';
+</script>
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:616]: I think it is safe to move the `smaily_rss_url_base` variable to the jQuery closure and right before the change callback:

```javascript
var smaily_rss_url_base = '<?php echo $smaily_rss_url_base; ?>';
$(".smaily-rss-options").change(function (event) {
    // ...
});
```

**Review by kaittodesk — CHANGES_REQUESTED** [review]

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl (hunk 1 above)
Comment (TomAbel) [reply upload/admin/view/template/extension/module/smaily_for_opencart.tpl:334]: Replaced with double quotes.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl (hunk 2 above)
Comment (TomAbel) [reply upload/admin/view/template/extension/module/smaily_for_opencart.tpl:339]: Not echoing out HTML

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl (hunk 3 above)
Comment (TomAbel) [reply upload/admin/view/template/extension/module/smaily_for_opencart.tpl]: Declaring it in controller

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl (hunk 4 above)
Comment (TomAbel) [reply upload/admin/view/template/extension/module/smaily_for_opencart.tpl]: Not echoing out HTML

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl (hunk 5 above)
Comment (TomAbel) [reply upload/admin/view/template/extension/module/smaily_for_opencart.tpl]: Moved it there

**Review by kaittodesk — CHANGES_REQUESTED** [review]
LGTM, but change the target branch name to `dev/2.3.x-1.4.0`.

**Review by kaittodesk — APPROVED** [review]

**Pull request merged**